test/Generators/Microsoft.Gen.Logging/Generated/Utils.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Compliance.Redaction;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Telemetry.Logging;
+using Microsoft.Extensions.Telemetry.Testing.Logging;
+using TestClasses;
+
+namespace Microsoft.Gen.Logging.Test;
+
+internal static class Utils
+{
+    private class Provider : ILoggerProvider
+    {
+        private readonly ILogger _logger;
+
+        public Provider(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public ILogger CreateLogger(string categoryName)
+        {
+            return _logger;
+        }
+
+        public void Dispose()
+        {
+            // nothing
+        }
+    }
+
+    public static (ILogger logger, FakeLogCollector collector) GetLogger(out FakeLogger fakeLogger)
+    {
+        fakeLogger = new FakeLogger();
+        using var provider = new Provider(fakeLogger);
+        var serviceCollection = new ServiceCollection();
+
+        serviceCollection.AddLogging(builder =>
+        {
+            builder.AddProvider(provider);
+            builder.EnableRedaction();
+        });
+
+        serviceCollection.AddRedaction(builder =>
+        {
+            builder.SetFallbackRedactor<StarRedactor>();
+        });
+
+        return (serviceCollection.BuildServiceProvider().GetRequiredService<ILoggerFactory>().CreateLogger(nameof(Utils)), fakeLogger.Collector);
```

Comment:
I don't really get this method. `FakeLogger` is an `ILogger`, right?
Do we really need to return `FakeLogCollector` as a second element in the tuple?
Can't we just return `fakeLogger`?

Reply:
Cleaned up GetLogger.